src/wl_init.c

```diff
@@ -94,8 +109,16 @@ static void pointerHandleEnter(void* data,
     if (!surface)
         return;
 
-    int focus = 0;
+    if (strcmp(*wl_proxy_get_tag((struct wl_proxy *) surface), proxy_tag)!=0)
```

Comment:
this should be null checked first

Reply:
Actually it is supposed to work like in the example of the commit https://cgit.freedesktop.org/wayland/wayland/commit/?id=493ab79bd2cdf8f9f12fb9e5522200dc668b85e0, i.e. just by comparing the addresses.

Reply:
I changed the linking of `proxy_tag`. This now works by comparing pointers and we do not have to check for NULL manually.